.github/workflows/r_valgrind.yml

```diff
@@ -1,8 +1,32 @@
 name: R valgrind tests
 
 on:
-  repository_dispatch:
-    types: [gha_run_r_valgrind]
+  workflow_dispatch:
+    inputs:
+      pr-branch:
+        type: string
+        description: |
+          branch the PR was submitted from.
+          Branches from forks should be prefixed with the user/org they originate from,
+          like '{user}:{branch}'.
+      pr-number:
+        type: string
+        description: Pull request ID, found in the PR URL.
+
+permissions:
```

Comment:
I used a few commits to find the minimal set of permissions needed for this workflow to run... I believe this is a tightly-scoped as is possible.

docs: https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#permissions